src/material.js

```diff
@@ -173,20 +173,42 @@ export class Material {
     }
 
     /**
-     * Calculates hash from basis and lattice. Algorithm expects the following:
+     * Calculates the hash for a material.
+     * If `isNonPeriodic` = false
+     * The hash is calculated from the basis and lattice. The algorithm expects the following:
      * - asserts lattice units to be angstrom
      * - asserts basis units to be crystal
      * - asserts basis coordinates and lattice measurements are rounded to hash precision
      * - forms strings for lattice and basis
      * - creates MD5 hash from basisStr + latticeStr + salt
+     *
+     * If `isNonPeriodic` = true
+     * The hash based on the `inchi` string for the material that is returned from getNonPeriodicHashMessage.
+     *
      * @param salt {String} Salt for hashing, empty string by default.
      * @param isScaled {Boolean} Whether to scale the lattice parameter 'a' to 1.
      */
     calculateHash(salt = '', isScaled = false) {
-        const message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        let message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        if (this._json.isNonPeriodic) {
+            message = this.getNonPeriodicHashMessage() || message;
```

Comment:
Why `II message`? We should probably raise an error if `isNonPeriodic` is set, but no inchi is supplied.